core/src/main/java/hudson/security/GlobalSecurityConfiguration.java

```diff
@@ -111,14 +111,10 @@ public boolean configure(StaplerRequest req, JSONObject json) throws hudson.mode
         // for compatibility reasons, the actual value is stored in Jenkins
         Jenkins j = Jenkins.getInstance();
         j.checkPermission(Jenkins.ADMINISTER);
-        if (json.has("useSecurity")) {
-            JSONObject security = json.getJSONObject("useSecurity");
-            j.setDisableRememberMe(security.optBoolean("disableRememberMe", false));
-            j.setSecurityRealm(SecurityRealm.all().newInstanceFromRadioList(security, "realm"));
-            j.setAuthorizationStrategy(AuthorizationStrategy.all().newInstanceFromRadioList(security, "authorization"));    
-        } else {
-            j.disableSecurity();
-        }
+
+        j.setDisableRememberMe(json.optBoolean("disableRememberMe", false));
+        j.setSecurityRealm(SecurityRealm.all().newInstanceFromRadioList(json, "realm"));
```

Comment:
It seems easy to lock oneself out by choosing None security realm, and almost any authorization strategy. Maybe we can check whether the security realm is set to the none realm, and if so, call `disableSecurity()`?

Form validation dependent on both fields would also be nice, if that's possible (don't remember right now).